spec/ama_layout/notifications/redis_store_spec.rb

```diff
@@ -0,0 +1,94 @@
+describe AmaLayout::Notifications::RedisStore do
+  subject do
+    described_class.new(
+      db: 4,
+      namespace: 'test_notifications',
+      host: 'localhost'
+    )
+  end
+
+  around(:each) do |example|
+    subject.clear
+    example.run
+    subject.clear
+  end
+
+  describe '#get' do
+    context 'when a key is not present' do
+      it 'returns nil' do
+        expect(subject.get('missing')).to be nil
+      end
+    end
+
+    context 'when a key is present' do
+      before(:each) do
+        subject.set('key', 'value')
+      end
+
+      it 'returns the value' do
+        expect(subject.get('key')).to eq('value')
+      end
+    end
+
+    context 'with a default value' do
+      it 'sets a nil key to the default value' do
+        subject.get('missing', default: 'test')
+        expect(subject.get('missing')).to eq('test')
+      end
+    end
+  end
+
+  describe '#set' do
+    it 'sets the value for a given key' do
+      subject.set('test', 'value')
+      expect(subject.get('test')).to eq('value')
+    end
+
+    it 'returns true' do
+      expect(subject.set('test', 'value')).to be true
+    end
+  end
+
+  describe '#delete' do
+    context 'when a value is deleted' do
+      before(:each) do
+        subject.set('key', 'value')
+      end
+
+      it 'returns true' do
+        expect(subject.delete('key')).to be true
+      end
+
+      it 'deletes the key' do
+        subject.delete('key')
+        expect(subject.get('key')).to be nil
+      end
+    end
+
+    context 'when a value is not deleted' do
+      it 'returns false' do
+        expect(subject.delete('key')).to be false
+      end
+    end
+  end
+
+  describe '#transaction' do
+    it 'does not commit if an exception is raised' do
+      begin
+        subject.transaction do |store|
+          store.set('key', 'value')
+          raise StandardError
```

Comment:
The `transaction` block prevents the key from getting written to if an exception is raised from within the block. So the key isn't really deleted in this case - it's not written yet.